src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

```diff
@@ -160,29 +147,75 @@ struct boss_kalecgos : public BossAI
         if (param == ACTION_ENRAGE || param == ACTION_ENRAGE_OTHER)
         {
             Talk(param == ACTION_ENRAGE ? SAY_KALEC_ENRAGE_SATH : SAY_SATH_ENRAGE_ME);
-            me->CastSpell(me, SPELL_CRAZED_RAGE, true);
-            events.CancelEvent(EVENT_CHECK_HEALTH);
+            DoCastSelf(SPELL_CRAZED_RAGE, true);
             return;
         }
         else if (param == ACTION_BANISH)
         {
-            me->CastSpell(me, SPELL_BANISH, true);
-            events.Reset();
+            DoCastSelf(SPELL_BANISH, true);
+            scheduler.CancelAll();
         }
         else if (param == ACTION_SATH_BANISH)
-            sathBanished = true;
+            _sathBanished = true;
         else if (param == ACTION_KALEC_DIED)
         {
-            events.Reset();
-            events2.ScheduleEvent(EVENT_TALK_BAD_1, 0);
+            scheduler.CancelAll();
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetReactState(REACT_PASSIVE);
+                me->CombatStop();
+                me->RemoveAllAuras();
+                me->SetUnitFlag(UNIT_FLAG_NON_ATTACKABLE);
+                Talk(SAY_EVIL_ENRAGE);
+            }, 1s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetDisableGravity(true);
+                me->HandleEmoteCommand(EMOTE_ONESHOT_LIFTOFF);
+            }, 4s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetVisible(false);
+                EnterEvadeMode();
+            }, 9s);
+
             ClearPlayerAuras();
             return;
         }
 
-        if (me->HasAura(SPELL_BANISH) && sathBanished)
+        if (me->HasAura(SPELL_BANISH) && _sathBanished)
         {
-            events.Reset();
-            events2.ScheduleEvent(EVENT_TALK_GOOD_1, 1000);
+            scheduler.CancelAll();
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetRegeneratingHealth(false);
+                me->RemoveAllAuras();
+                me->SetReactState(REACT_PASSIVE);
+                me->CombatStop();
+                me->SetUnitFlag(UNIT_FLAG_NON_ATTACKABLE);
+                me->SetFaction(FACTION_FRIENDLY);
+            }, 1s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                if (Creature* Sath = instance->GetCreature(DATA_SATHROVARR))
+                {
+                    summons.Despawn(Sath);
+                    Unit::Kill(me, Sath);
+                }
+            }, 2s);
+
+            Talk(SAY_GOOD_PLRWIN, 10s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetDisableGravity(true);
+                me->HandleEmoteCommand(EMOTE_ONESHOT_LIFTOFF);
+            }, 15s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->SetVisible(false);
+                me->KillSelf();
+            }, 20s);
```

Comment:
This was 30s

Reply:
Yes, I changed all of the timers, because they’re not cascaded (they’re concurrent and not one inside the other, not sure if it makes sense, could probably explain better later)

Reply:
Alright I re check later. I had added up all the cascaded timings, too and came to a different result

Reply:
Yes, because as I said, I changed the timers

Reply:
But this means that this event now triggers 10s earlier than in the previous script version imo. But I'll also take another look, maybe I didnt add up all the cascaded timers correctly.

Reply:
So yeah if you wanted those events to trigger earlier than this is fine.

Reply:
yes